src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -189,6 +222,34 @@ public static RequestDelegateResult Create(MethodInfo methodInfo, Func<HttpConte
         return HandleRequestBodyAndCompileRequestDelegate(responseWritingMethodCall, factoryContext);
     }
 
+    private static Func<RouteHandlerFilterContext, ValueTask<object?>> CreateFilterPipeline(MethodInfo methodInfo, Expression? target, FactoryContext factoryContext)
+    {
+        // httpContext.Response.StatusCode == 400
+        // ? Task.CompletedTask
+        // : handler((string)context.Parameters[0], (int)context.Parameters[1])
+        var filteredInvocation = Expression.Lambda<Func<RouteHandlerFilterContext, ValueTask<object?>>>(
+            Expression.Condition(
+                Expression.Equal(FilterContextHttpContextStatusCodeExpr, Expression.Constant(400)),
+                CompletedValueTaskExpr,
+                Expression.Block(
+                    new[] { TargetExpr },
+                    Expression.Call(WrapObjectAsValueTaskMethod,
+                        target is null
+                            ? Expression.Call(methodInfo, factoryContext.ContextArgAccess)
+                            : Expression.Call(target, methodInfo, factoryContext.ContextArgAccess))
+                )),
+            FilterContextExpr).Compile();
+
+        for (int i = factoryContext.Filters!.Count - 1; i >= 0; i--)
```

Comment:
```suggestion
        for (var i = factoryContext.Filters!.Count - 1; i >= 0; i--)
```

I wasn't fully convinced I should send a PR to turn on codeanalysis warning for not using vars, but now I feel differently.